api/src/policies/datasets-policy.ts

```diff
@@ -97,10 +97,16 @@ export class DatasetsPolicy extends BasePolicy<Dataset> {
             ],
           },
         },
-      })
+      }
     }
 
-    return modelClass.scope({ method: ["accessibleViaAccessGrantsBy", user] })
+    return {
+      where: {
+        id: {
+          [Op.in]: datasetsAccessibleViaAccessGrantsByUserQuery,
+        },
+      },
+    }
```

Comment:
Inlining code is the least painful solution to needing to use a scope in another scope.